### ISSUE TITLE
Invalid method name in metrics classes

### ISSUE THREAD

**Issue by markustoivonen** [tensorflow/addons#2493]
**System information**
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): Mac OS X 11.2.3.
- TensorFlow version and how it was installed (source or binary): 2.5.0, binary (pip)
- TensorFlow-Addons version and how it was installed (source or binary): 0.13.0, binary (pip)
- Python version: 3.7.
- Is GPU used? (yes/no): no

**Describe the bug**

Not a bug but, a warning is raised when using the addon metrics classes that are inherited from the `Metric`class. The `reset_states` method should be called `reset_state`.

**Other info / logs**

Relevant warning message:

`UserWarning: Metric F1Score implements a 'reset_states()' method; rename it to 'reset_state()' (without the final "s"). The name 'reset_states()' has been deprecated to improve API consistency.`

The warning is raised [here](https://github.com/tensorflow/tensorflow/blob/431e288e4d4ce4af4664175eaaa6e68b12500079/tensorflow/python/keras/metrics.py#L247)

**Comment by markustoivonen** [comment]
Oh, there was an open PR about this. Sorry about that.
